src/Serialization/Fields.jl

```diff
@@ -48,3 +48,132 @@ function load_internal(s::DeserializerState, ::Type{gfp_fmpz_elem}, dict::Dict)
     F = load_type_dispatch(s, Nemo.GaloisFmpzField, dict[:parent])
     return F(load_type_dispatch(s, fmpz, dict[:data]))
 end
+
+################################################################################
+# SimpleNumField
+function save_internal(s::SerializerState, K::SimpleNumField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K)),
+        :var => save_type_dispatch(s, String(var(K)))
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{<: SimpleNumField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    var = load_type_dispatch(s, dict[:var], check_namespace=false)
+    K, _ = NumberField(def_pol, var, cached=false)
+
+    return K
+end
+
+################################################################################
+# FqNmodfinitefield
+function save_internal(s::SerializerState, K::FqNmodFiniteField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K))
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{FqNmodFiniteField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
```

Comment:
I think we always know the type here, so you could write
```suggestion
    def_pol = load_type_dispatch(s, gfp_poly, dict[:def_pol])
```
But perhaps you would like to extend this method to cover more finite field implementations, e.g. `FqFiniteField` (created e.g. via `GF(fmpz(2), 2)`). Then you could use something like `dense_poly_type(elem_type(T))` to figure out the expected type (where `T` is the type of your ring, so change the second argument to something like `T::Type{Union{FqNmodFiniteField, FqFiniteField}}` -- though all of this is untested